presto-docs/src/main/sphinx/optimizer/logical-planner.rst

```diff
@@ -0,0 +1,35 @@
+*****************
+Logical Planner
+*****************
+
+Presto currently has a rule based optimizer. A given SQL query goes through syntactic
+analysis, semantic analysis, logical plannning, physical planning and then execution.
+Syntax analysis or parsing is used to determine various structures in the given
+query and if they conform to the given rules. Next step is semantic analysis which
+deals with type checking or things like verifying that the column names used in the query exist for a given table.
+The logical planner then takes this syntactic and semantic information for a query and using metadata
+and session information creates a base logical plan.
+
+Consider this example,
+
+.. code-block:: sql
+
+    SELECT nationkey FROM nation JOIN region ON nation.regionkey = region.regionkey
+
+For this query Presto will generate a logical base plan,
+
+.. code-block:: none
+
+            ProjectNode
+                |
+            JoinNode
+                /   \
+    TableScanNode   TableScanNode
+    (nation)        (region)
+
+
+Each of the node classes in this logical plan are derived from PlanNode. Base logical plan is then
```

Comment:
Also, grammar nit: "of the node classes in this logical plan" is a prepositional phrase, and does not affect the pluralness of "Each", which is singular. It should read Each ... is derived from ....